### ISSUE TITLE
rsync

### ISSUE THREAD

**Issue by nihil-admirari** [holzschu/a-Shell-commands#25]
Is it possible to add rsync to a-Shell that can synchronise with a USB flash drive?

`pickFolder` allows one to `cd` to a folder on a USB drive, but for `rsync` at least one of the source or the destination must be accessible as a command line argument; they can't both be current directories.

**Comment by holzschu** [comment]
We don't have rsync. On the other hand, once you have selected a folder with pickFolder, it can be used as an argument on the command line: all folders added with pickFolder are added as bookmarks, and can be addressed with ~bookmarkName. 

showmarks will show you the list of existing bookmarks.

**Comment by nihil-admirari** [comment]
> We don't have rsync.

I know that there's no `rsync` currently; it's kind of a feature request. If it's not possible to have `rsync`, or if you're just not going to implement `rsync`, you can simply close the issue.

> On the other hand, once you have selected a folder with pickFolder, it can be used as an argument on the command line: all folders added with pickFolder are added as bookmarks, and can be addressed with ~bookmarkName.

Thanks, didn't know it's possible.

**Comment by weiszr** [comment]
To add my support for this request: It would be great to add rsync in order sync a director from a remote machine to linked ipad storage via pickFolder. I think this would enhance the use cases for the app.

**Comment by papypaprika** [comment]
I concur, rsync would be a great addition to the packages available with a-shell.

**Comment by holzschu** [comment]
Please do. You know the TestFlight URL already, I think, but just in case: https://testflight.apple.com/join/WUdKe3f4

**Comment by weiszr** [comment]
It is crashing...on iPadOS26.1. I looked at the log..not very informative.


______________________________________________________

Dr. Robert Weiss

Academy of Integrated Science, Director

Professor of Natural Hazards, Department of Geoscience

College of Science



Contact:

Offices: 3066A Derring Hall

Phone:+1-540-231-2334<tel:+1-540-231-2334>

Fax: +1-540-231-3386<tel:+1-540-231-3386>

Twitter: @CoastalHazardVT



Web

Academy: Academy of Integrated Science<https://www.ais.science.vt.edu/>

________________________________

**Comment by weiszr** [comment]
It is working now without issues.

I usually use -avhu and --progress. I noticed that there is "u" option which prevents overwriting newer files at the destination. Is that just not included?

Good work though!!!

Robret


______________________________________________________

Dr. Robert Weiss

Academy of Integrated Science, Director

Professor of Natural Hazards, Department of Geoscience

College of Science



Contact:

Offices: 3066A Derring Hall

Phone:+1-540-231-2334<tel:+1-540-231-2334>

Fax: +1-540-231-3386<tel:+1-540-231-3386>

Twitter: @CoastalHazardVT



Web

Academy: Academy of Integrated Science<https://www.ais.science.vt.edu/>

________________________________

**Comment by holzschu** [comment]
Yes, it's the BSD version (openrsync, https://www.openrsync.org) instead of the GNU version (rsync, https://rsync.samba.org). They behave mostly the same way, but `--progress` is definitely missing.
